src/ActionList/ActionList.features.stories.tsx

```diff
@@ -96,6 +96,146 @@ export const WithVisualListHeading = () => (
   </ActionList>
 )
 
+export const ListWithNewGroupHeadingAPI = () => (
+  <ActionList>
+    <ActionList.Heading as="h2">List Heading</ActionList.Heading>
+    <ActionList.Group data-test-id="ActionList.Group">
+      <ActionList.GroupHeading as="h3">Group 1 Heading</ActionList.GroupHeading>
+      <ActionList.Item onClick={() => {}}>
+        <ActionList.LeadingVisual>
+          <FileDirectoryIcon />
+        </ActionList.LeadingVisual>
+        app/assets/modules
+      </ActionList.Item>
+      <ActionList.Item onClick={() => {}}>
+        <ActionList.LeadingVisual>
+          <FileDirectoryIcon />
+        </ActionList.LeadingVisual>
+        src/react/components
+      </ActionList.Item>
+      <ActionList.Item onClick={() => {}}>
+        <ActionList.LeadingVisual>
+          <FileDirectoryIcon />
+        </ActionList.LeadingVisual>
+        memex/shared-ui/components
+      </ActionList.Item>
+      <ActionList.Item onClick={() => {}}>
+        <ActionList.LeadingVisual>
+          <FileDirectoryIcon />
+        </ActionList.LeadingVisual>
+        views/assets/modules
+      </ActionList.Item>
+    </ActionList.Group>
+
+    <ActionList.Group>
+      <ActionList.GroupHeading as="h3">Group 2 Heading</ActionList.GroupHeading>
+      <ActionList.Item onClick={() => {}}>
```

Comment:
I'm not sure what our conventions are, but when would we use an empty handler for events versus using storybook actions?

Reply:
This is a really good question. I am not really sure either. I find storybook actions useful when the story is about the action like the [data table action stories](https://primer.style/react/storybook/?path=/story/drafts-components-datatable-features--with-row-action) you added are great examples for this use case in my opinion. The stories I added or the ones I mentioned [here](https://github.com/primer/react/pull/3900#discussion_r1385803143) to update are not focusing on actions, they are focusing on semantics, visuals, and having group titles. This is why, I didn't worry about the action itself but I am more than happy to update them to use storybook actions if it is going to be helpful. 
Re-thinking about it now, it is probably helpful to log something (storybook actions could be very beneficial in this case) when the click event is trigger at least so empty event handler isn't very helpful here. 

What are your thoughts??